scripts/zephyr_module.py

```diff
@@ -389,7 +389,12 @@ def git_revision(path):
             project_path = PurePath(project.posixpath).as_posix()
             revision, dirty = git_revision(project_path)
             workspace_dirty |= dirty
-            if project.sha(MANIFEST_REV_BRANCH) != revision:
+            try:
+                manifest_rev_sha = project.sha(MANIFEST_REV_BRANCH)
```

Comment:
This executes only if `west_projs is not None`, so it assumes you are using west? Are you adding extra non-west projects via cmd line?

Reply:
nope, currently everything is managed by west

Reply:
Then why are you missing a `manifest-rev` branch?

Reply:
I have a project structure roughly resembling this:
```
project/
├── firmware
│   ├── application_firmware/ # .git/ project with application firmware
│   │   ├── CMakeLists.txt
│   │   ├── prj.conf
│   │   ├── west.yml        # main manifest with optional import(s) and override(s)
│   │   └── src/
│   │       └── main.c
│   └── components/  # second project with more custom code
├── zephyr/                 # .git/ project
│   └── west.yml            # This can be partially imported with lower precedence or ignored.
...
```
`application_firmware` houses my main application, `components` represents a second repository (defined as a module) where I also develop concurrently. Both repositories do not have a `manifest-rev`.
(indeed, I could add the `manifest-rev` to `components`, but the branch wouldn't be meaningful and just be dangling somewhere around).

Reply:
Does `components` have its own `zephyr/module.yml` and is it a project referenced by `application_firmware/west.yml`? Because if that is the case, it must have a `manifest-rev`. Regarding the manifest repo itself (AFAICT `application_firmware` in your case, that one does not have `manifest-rev` but has special treatment in the code you changed already in `main`.

Reply:
It has by default, but after alot of commits it basically dangled somewhere around. I do seldom run `west update`, most of the time only if I really need an update. Of course all CI jobs have the `manifest-rev`.
The sole purpose of the change is to allow repositories where development happens a more relaxed approach with the branch naming.

If you think this is a bad idea, I will change the code that at least the error message is meaningful.